Sources/StreamChat/Models/Member.swift

```diff
@@ -136,10 +149,7 @@ public enum MemberRole: String, Codable, Hashable {
         case "owner":
             self = .owner
         default:
-            throw DecodingError.dataCorruptedError(
-                in: container,
-                debugDescription: "MemberRole string value `\(value)` doesn't match any of the known roles"
-            )
+            self = MemberRole(rawValue: value)
```

Comment:
Maybe this is actually only what we need? 🤔 We don't need a switch case. 

If we need to compare roles, we would do `user.memberRole == .moderator`, this will always compare the raw value. So there is no need for the switch case. The static props is just used to be easier to compare with known roles, or to create easily roles .

Reply:
I think we need the switch case as we map two values i.e. "channel_moderator" and "moderator" to the role `moderator`?

Reply:
Why do we have those two types tho ? 🤔 are they really the same role?

Reply:
Yes, we do need to map those two to the same role